pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -414,3 +434,124 @@ func (c *EtcdProxyController) handleObject(obj interface{}) {
+	if errors.IsNotFound(err) {
+		data := make(map[string]string, 1)
+		data["ca.pem"] = EtcdProxyCACert
+		caConfigMap, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).
+			Create(newConfigMap(etcdstorage, etcdProxyCAConfigMapName(etcdstorage), c.config.ControllerNamespace, data))
+	}
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	if !metav1.IsControlledBy(caConfigMap, etcdstorage) {
+		caConfigMap.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+		caConfigMap, err = c.kubeclientset.CoreV1().ConfigMaps(c.config.ControllerNamespace).Update(caConfigMap)
+		if err != nil {
+			msg := fmt.Sprintf(MessageErrResourceReclaimed, caConfigMap.Name)
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceReclaimed, msg)
+			return err
+		}
+		msg := fmt.Sprintf(ResourceReclaimedReason, caConfigMap.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ResourceReclaimed, msg)
+	}
+
+	return nil
+}
+
+// createEtcdProxyCerts creates Secret in controller namespace with Etcd Proxy Server certificate and key.
+func (c EtcdProxyController) createEtcdProxyCerts(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
+	// Secret for the etcd proxy server certs in controller namespace.
+	certsSecret, err := c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+		Get(etcdProxyServerCertsSecret(etcdstorage), metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		data := make(map[string]string, 2)
+		data["server.pem"] = EtcdProxyServerCert
+		data["server-key.pem"] = EtcdProxyServerKey
+		certsSecret, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).
+			Create(newSecret(etcdstorage, etcdProxyServerCertsSecret(etcdstorage), c.config.ControllerNamespace, data))
+	}
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+	if !metav1.IsControlledBy(certsSecret, etcdstorage) {
+		certsSecret.SetOwnerReferences([]metav1.OwnerReference{
+			*metav1.NewControllerRef(etcdstorage, etcdstoragev1alpha1.SchemeGroupVersion.WithKind("EtcdStorage")),
+		})
+		certsSecret, err = c.kubeclientset.CoreV1().Secrets(c.config.ControllerNamespace).Update(certsSecret)
+		if err != nil {
+			msg := fmt.Sprintf(MessageErrResourceReclaimed, certsSecret.Name)
+			c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceReclaimed, msg)
+			return err
+		}
+		msg := fmt.Sprintf(ResourceReclaimedReason, certsSecret.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ResourceReclaimed, msg)
+	}
+
+	return nil
+}
+
+// updateEtcdProxyAPIServerCACert updates the ConfigMap in the aggregated API server namespace with the CA certificate.
+func (c *EtcdProxyController) updateEtcdProxyAPIServerCACert(etcdstorage *etcdstoragev1alpha1.EtcdStorage) error {
```

Comment:
the EtcdProxy in the func name is confusing IMO. updateAPIServerCACertConfigMap maybe?